File: 0301_0400/0361/0361.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    s = f.readline().strip()
    d, i, n = 0, 0, len(s)
    while d < n - i:
        d = max(d, s.rindex(s[i]) - i)
        i += 1
    q.write(str(d))

File: 0301_0400/0361/0361.c
#include <stdio.h>
#include <string.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int d;
	char s[101];

	fscanf(f, "%s", s);
	d = 0;
	for (int i = 0, n = strlen(s), k; d < n - i; ++i)
		if ((k = strrchr(s, s[i]) - s - i) > d)
			d = k;
	fprintf(q, "%d", d);
	return 0;
}
